crates/verify/src/bytecode.rs

```diff
@@ -0,0 +1,557 @@
+        let mut dict = figment::value::Dict::new();
+        if let Some(block) = &self.block {
+            dict.insert("block".into(), figment::value::Value::serialize(block)?);
+        }
+        if let Some(rpc_url) = &self.rpc_url {
+            dict.insert("eth_rpc_url".into(), rpc_url.to_string().into());
+        }
+        dict.insert("verification_type".into(), self.verification_type.to_string().into());
+
+        // if let Some(root) = self.root.as_ref() {
+        //     dict.insert("root".to_string(), figment::value::Value::serialize(root)?);
+        // }
+        Ok(figment::value::Map::from([(Config::selected_profile(), dict)]))
+    }
+}
+
+impl VerifyBytecodeArgs {
+    /// Run the `verify-bytecode` command to verify the bytecode onchain against the locally built
+    /// bytecode.
+    pub async fn run(mut self) -> Result<()> {
+        // Setup
+        let config = self.load_config_emit_warnings();
+        let provider = ProviderBuilder::new(&config.get_rpc_url_or_localhost_http()?).build()?;
+        println!(
+            "Verifying bytecode for contract {} at address {}",
+            Paint::green(self.contract.name.clone()),
+            Paint::green(self.address.to_string())
+        );
+        // If chain is not set, we try to get it from the RPC
+        // If RPC is not set, the default chain is used
+        let chain = match config.get_rpc_url() {
+            Some(_) => {
+                let chain_id = provider.get_chain_id().await?;
+                // Convert to u64
+                Chain::from(chain_id.to::<u64>())
+            }
+            None => config.chain.unwrap_or_default(),
+        };
+
+        // Set Etherscan options
+        self.etherscan_opts.chain = Some(chain);
+        self.etherscan_opts.key =
+            config.get_etherscan_config_with_chain(Some(chain))?.map(|c| c.key);
+        // Create etherscan client
+        let etherscan = Client::new(chain, self.etherscan_opts.key.clone().unwrap())?;
+
+        // Get the constructor args using `source_code` endpoint
+        let source_code = etherscan.contract_source_code(self.address).await?;
+
+        let constructor_args = match source_code.items.first() {
+            Some(item) => item.constructor_arguments.clone(),
+            None => {
+                eyre::bail!("No source code found for contract at address {}", self.address);
+            }
+        };
+        let provided_constructor_args = if let Some(args) = self.constructor_args.to_owned() {
+            args
+        } else if let Some(path) = self.constructor_args_path.to_owned() {
+            // Read from file
+            let res = read_constructor_args_file(path)?;
+            // Convert res to Bytes
+            res.join("")
+        } else {
+            constructor_args.to_string()
+        };
+        if provided_constructor_args != constructor_args.to_string() {
+            eyre::bail!("Constructor args do not match, verification will fail.");
+        }
```

Comment:
what's the point of providing constructor args through CLI if those are required to exactly match args from etherscan?

Reply:
Idea is to cross-check everything locally with everything onchain. This check pinpoints difference constructor_args. Similarly, also implemented `fn find_mismatch_in_settings` which notifies the user about the difference in compiler settings. 

Currently, bailing out on encountering differing constructor_args but could instead notify the user and continue with check like `find_mismatch_in_settings`

Reply:
imo a better UX would be to avoid taking user input where possible, fetch both constructor args and compiler settings from Etherscan, compile, and then display whether we have a match with any local contracts along with constructor arguments (abi decoded?) and compilation settings.

@mds1 what do you think?

Reply:
Agreed with @klkvr, we should make UX as easy as possible. So we can fetch constructor args from etherscan by default unless the user explicitly provided constructor args as a CLI arg. 

We probably can fetch solc settings also, but I don't know if Etherscan exposes all the correct data in the API. I know the UI does not show if via-ir was used for example, but maybe the API provides more info. If the API does provide sufficient info we can use that, otherwise fallback to the settings of the profile being used

Reply:
> The API doesn't provider anything more than what can be seen on the UI.

Got it, so for solc config, means we should rely on the a profile since the API is not comprehensive 

> And we are already using the args from etherscan by default. Instead of throwing an error, Now notifying the user that the args differ.

Sorry what do you mean here? If we are using args from etherscan, we don't need the user to provide them, and don't need to warn/error about different args right? I might be misunderstanding something

Reply:
> Sorry what do you mean here? If we are using args from etherscan, we don't need the user to provide them, and don't need to warn/error about different args right? I might be misunderstanding something

Two things here: 
Firstly, providing constructor args is optional for the user. 
Second, we're always fetching the constructor_args from etherscan regardless of whether the user provides them or not. 

In light of the above:
- We always use the constructor_args from etherscan by default while cross-checking the locally built bytecode with the onchain one.
- If the user has provided constructor_args, then we compare them with the one retrieved from etherscan and notify the user if they differ. I thought this notification was useful since it tells the user that their local copy differs from what was used to deploy the onchain one. Along the same lines, we're notifying about differing compiler settings using `fn find_mismatch_in_settings`. This notification was to idiot-proof the UX so that we could pinpoint each case in which the local bytecode could differ. But if you think this is not helpful, I'll remove it.

Reply:
Got it! That makes sense, thanks for the explanation. I'm onboard with that. Agree with your suggestion to only warn (instead of error) if they do not match, since to your point there is a chance that the user's version is correct